bql/grammar/grammar_test.go

```diff
@@ -113,6 +113,16 @@ func TestAcceptByParse(t *testing.T) {
 		                          /room<000> "connects_to"@[] /room<001>};`,
 		`delete data from ?world {/room<000> "named"@[] "Hallway"^^type:text.
 		                          /room<000> "connects_to"@[] /room<001>};`,
+		// Test Construct clause.
+		`construct {?s "foo"@[,] ?o} into ?a from ?b where {?s "foo"@[,] ?o} having ?s = ?o;`,
```

Comment:
This is exactly the point I was making above. If you want to create triples for each of the `foo` predicates with the right timestamp, you could write it like this instead

```
CONSTRUCT {?s ?p ?s} 
INTO ?a 
FROM ?b 
WHERE {
  ?s "foo"@[,] as ?p ?s
} ;
```

Also see how you can also avoid the having by properly playing with the binding, but that is beside the point here, since we should definitely have test that cover the having clause :smile: